src/Pos/Util.hs

```diff
@@ -250,12 +252,12 @@ forcePutMVar mvar val = do
 -- satisfies, it is returned.
 readMVarConditional :: (MonadIO m) => (x -> Bool) -> MVar x -> m x
 readMVarConditional predicate mvar = do
-    rData <- liftIO . readMVar $ mvar -- first we try to read for optimization only
+    rData <- readMVar mvar -- first we try to read for optimization only
     if predicate rData then pure rData
     else do
-        tData <- liftIO . takeMVar $ mvar -- now take data
```

Comment:
Don't bother with `liftIO`, it will be cleaned up when I push my hlint PR (in fact, it makes my work a bit harder because of merge conflicts :) )